### PR TITLE
[CORE-287] Generate Release Tags with Travis

### PR DESCRIPTION
**Pull request by lexn82**
- Put Travis release and tagging logic into one file to simplify maintencance
- Added auto-tagging releases in Git

### CODE REVIEW

**Comment by cdegroot** [comment]
LGTM (if it works - haven't tested that)

File: .travis.release.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
```
Comment (divtxt) [inline .travis.release.sh]: I recommend

    set -e  # fail on errors

**Comment by divtxt** [comment]
:+1:

File: .travis.release.sh (hunk 1 above)
Comment (lexn82) [inline .travis.release.sh]: I think some of these are OK to fail, except the last line, which is combined with `&&` to ensure consistency.

**Pull request merged**